### PR TITLE
add support for targeting easyblocks and framework in --new-pr (REVIEW)

### PR DESCRIPTION
**Pull request by migueldiascosta**
wrapper PR for #1876 - thanks @Caylo

### CODE REVIEW

**Comment by ocaisa** [comment]
@boegel Too much history in this one for me to review easily :(

File: easybuild/tools/github.py [hunk 1]
```diff
@@ -974,6 +983,85 @@ def find_software_name_for_patch(patch_name, ec_dirs):
...
+        'paths_in_repo': [],
+        'new': [],
+    }
+
+    subdir = os.path.join('easybuild', 'easyblocks')
+    if os.path.exists(os.path.join(target_dir, subdir)):
+        for path in paths:
+            cn = get_easyblock_class_name(path)
+            if not cn:
+                raise EasyBuildError("Invalid easyblock file")
```
Comment (akesandgren) [inline easybuild/tools/github.py]: This should probably print the name of the file.

File: easybuild/tools/github.py [hunk 2]
```diff
@@ -974,6 +983,85 @@ def find_software_name_for_patch(patch_name, ec_dirs):
...
+
+            eb_name = remove_unwanted_chars(decode_class_name(cn).replace('-', '_')).lower()
+            if is_generic_easyblock(cn):
+                target_path = os.path.join(subdir, GENERIC_EB, "%s.%s" % (eb_name.lower(), PYTHON_EXTENSION))
```
Comment (akesandgren) [inline easybuild/tools/github.py]: eb_name is already lowered

File: easybuild/tools/github.py [hunk 3]
```diff
@@ -974,6 +983,85 @@ def find_software_name_for_patch(patch_name, ec_dirs):
...
+            else:
+                letter = os.path.basename(path).lower()[0]
```
Comment (akesandgren) [inline easybuild/tools/github.py]: Should this use path or eb_name??

File: easybuild/tools/github.py [hunk 4]
```diff
@@ -974,6 +983,85 @@ def find_software_name_for_patch(patch_name, ec_dirs):
...
+                target_path = os.path.join(subdir, letter, "%s.%s" % (eb_name, PYTHON_EXTENSION))
+
+            full_target_path = os.path.join(target_dir, target_path)
+            file_info['eb_names'].append(eb_name)
+            file_info['paths_in_repo'].append(full_target_path)
+            file_info['new'].append(not os.path.exists(full_target_path))
+            copy_file(path, full_target_path, force_in_dry_run=True)
+
+    else:
+        raise EasyBuildError("Subdir easyblocks not found")
```
Comment (akesandgren) [inline easybuild/tools/github.py]: It's usually better to print the full path that wasn't found.

File: easybuild/tools/github.py (hunk 3 above)
Comment (migueldiascosta) [reply easybuild/tools/github.py]: good question. I guess eb_name is more consistent.

**Comment by migueldiascosta** [comment]
thanks @boegel, this looks much better!

**Pull request merged**